src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -696,9 +696,17 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                     handleOne(callbacks, idx + 1);
                 } else if (callback instanceof AuthorizeCallback) {
                     final AuthorizeCallback authorizeCallback = (AuthorizeCallback) callback;
+                    String authenticationID = authorizeCallback.getAuthenticationID();
+                    if (authenticationID == null) {
+                        authenticationID = getAuthenticationPrincipal().getName();
```

Comment:
I understand; the problem is that there might not be any authenticated principal, or if there is, re-setting it from the principal name might result in a conflicting name which will lead to an error.